### ISSUE TITLE
Langchain nodes not respecting HTTP proxy settings

### ISSUE THREAD

**Issue by rasmusson** [n8n-io/n8n#10901]
### Bug Description

When using a Langchain models like OpenAI and Gemini, HTTP calls going to the models online does not use the HTTP proxy defined for N8N. But go straight to the internet. 

I found this as my N8N is not allowed to access internet directly. Allowing direct access resolves the issue

### To Reproduce

1. Setup n8n to use a HTTP proxy using environmental variables HTTP_PROXY and HTTPS_PROXY.
2. Turn off access to internet except through proxy
3. Set up and run a simple AI flow using chat trigger, conversational agent and OpenAI model.

### Expected behavior

Preferably, the AI model nodes should use the proxy defined for n8n. If not there should be a possibility to configure proxy separately for these nodes

### Operating System

Docker

### n8n Version

1.59.3

### Node.js Version

Provided by docker image

### Database

SQLite (default)

### Execution mode

main (default)

**Comment by Joffcom** [comment]
Hey @rasmusson,

We have created an internal ticket to look into this which we will be tracking as "GHC-244"

**Comment by Joffcom** [comment]
Hey @rasmusson,

I suspect this is not specific to the Langchain nodes and is likely linked to another general issue that we have open where Axios the http package we use doesn't pick up system defined proxies correctly if it is an HTTP proxy due to a failure on the CONNECT method.

**Comment by rasmusson** [comment]
Ok. As I understand the axios issue resulted in bad gateway error. In my case in the langchain nodes I dont get any error, the request just go straight to internet without going through the proxy. The node works just fine if I allow direct internet access but keep the system defined proxy.

**Comment by Joffcom** [comment]
Hey @rasmusson,

That sounds a bit different, How are you setting the system proxy?

**Comment by rasmusson** [comment]
yes, http_proxy and https_proxy env vars. Im using information extractor together with open AI model

**Comment by Joffcom** [comment]
Perfect, I have moved this to our AI team to investigate. It looks like Langchain may have an issue with proxies.

**Comment by pemontto** [comment]
This seems even more broken in recent versions:
In 1.50.2 I can at least run AI model nodes with a firewall rule bypassing the proxy 😢 (the rest of the traffic correctly goes via the proxy)
![image](https://github.com/user-attachments/assets/48846688-bda5-4c5d-b501-c4ff4c90dcb8)

However in 1.69.2 the AI Agent never seems to call the chat model, it will sit there indefinitely. Even debug logging gives no indication why. I've tried adding the model base url to `no_proxy` but it still fails. The only way it will work is by removing `http_proxy` and `https_proxy` entirely... which means we can have AI and nothing else, or everything else but not AI 😖
![image](https://github.com/user-attachments/assets/0da38593-0c28-4ac6-a82e-3c356e35108a)

Debug logging from 1.69.2
![image](https://github.com/user-attachments/assets/c53513b1-e3cd-40c7-ab5a-76d6d1e5c0bb)

**Comment by alparo** [comment]
I confirm. Even HTTP Request node doesn't work with globally set proxy:
```
    environment:
...
      - HTTP_PROXY=http://v2raya:20171
      - HTTPS_PROXY=http://v2raya:20171
```

I get:
```
Service unavailable - try again later or consider setting this node to retry automatically (in the node settings)
503 - ""
```
n8n version: 1.74.3

**Comment by pemontto** [comment]
@alparo that doesn't seem like an error reaching the proxy. Confirming I'm on `1.74.3` with these in the environment
```shell
http_proxy=http://proxy.example.com:3128
https_proxy=http://proxy.example.com:3128
```

And my normal HTTP requests are going via the proxy successfully... albeit not using HTTP CONNECT (another axios issue)

<img width="1189" alt="Image" src="https://github.com/user-attachments/assets/d0e57bfa-da0d-4e2c-b076-3eed3bfa2d69" />

**Comment by jasonliusz** [comment]
Currently v1.80.5 still repro this issue. 
I use Charles as Proxy to inspect the payload, google mail node can work via Charles proxy, but ai agent node can't. 
Both nodes can get data correctly, just the later one cannot using the proxy.
Through Charles, I can see that google mail api is called through User Agent AIOS, it proved that ai agent may not use the same   http connection tunnel.

**Comment by f1h0** [comment]
Dear experts Can you please tell me how to use proxy in openAI node? It's already the fourth day of agonizing some more solution to make requests

**Comment by bhavesh0124** [comment]
Still facing the same issue, the proxy doesn't work in Gemini, Mistral, openai, any of the chatmodels

**Comment by f1h0** [comment]
I'm sharing my secrets with you. I ran out of patience. In my despair, I pulled out all my hair, and now I'm bald. I look like Jim Carrey emerging from an elephant's ass.

So, what did I do? I decided not to struggle with these proxies. I changed my location to an available IP address that allows me to work directly with any AI agent nodes.

**Comment by bhavesh0124** [comment]
I better start looking for alternatives after your wild experience.

I'm unfortunately behind  a corporate proxy, and there is no way i can bypass them.

**Comment by f1h0** [comment]
Perhaps someone has good experience using Docker Traefik Proxy.  
It's quite a powerful tool that allows traffic to be directed according to various rules.  

Maybe there's a configuration that can be linked with N8N.

**Comment by bhavesh0124** [comment]
By the way, i added a piece of code in request-helper-functions.js from this thread:
https://community.n8n.io/t/n8n-fails-to-connect-to-the-internet-via-corporate-proxy/81167/33

And it made my http node working, but I guess these AI nodes somehow have different configurations somewhere

**Comment by hemlet** [comment]
same problem. 

I deployed n8n on a hongkong vps, added http and https proxy to n8n environment, when I use http node to call gemini api, it works. but if I directly use gemini model node,  it failed when I add gemini credential, n8n just show bad request, seems like google node just use proxy config in n8n environment.

**Comment by Marshall-Qiu** [comment]
> same problem.
> 
> I deployed n8n on a hongkong vps, added http and https proxy to n8n environment, when I use http node to call gemini api, it works. but if I directly use gemini model node, it failed when I add gemini credential, n8n just show bad request, seems like google node just use proxy config in n8n environment.

Hi  @Joffcom  same issue here, failed when using gemini credential, n8n show bad request even with proper proxy setting (open ai model works), could you help to take a look ?
